src/lib/Config/CredentialsResolver.php

```diff
@@ -12,29 +12,29 @@
 
 abstract class CredentialsResolver implements CredentialsResolverInterface
 {
-    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
-    protected $configResolver;
+    protected ConfigResolverInterface $configResolver;
 
     public function __construct(ConfigResolverInterface $configResolver)
     {
         $this->configResolver = $configResolver;
     }
 
-    abstract protected function getRequiredCredentials(?string $siteAccess = null): array;
-
     /**
-     * {@inheritdoc}
+     * @return array<string>
      */
+    abstract protected function getRequiredCredentials(?string $siteAccess = null): array;
+
     public function hasCredentials(?string $siteAccess = null): bool
     {
         $credentials = $this->getRequiredCredentials($siteAccess);
 
-        foreach ($credentials as $credentialKey => $credentialValue) {
-            if (empty($credentialValue)) {
-                return false;
+        $emptyCredentials = array_filter(
+            $credentials,
+            static function (?string $credential = null): bool {
+                return empty($credential);
```

Comment:
What is the reason to change original fail-fast approach ?

Reply:
> What is the reason to change original fail-fast approach ?

Logic seems to be incorrect.

In the original code the method returned `false` when encountering an empty credential entry.

In the new code method returns `false` only if all entries are empty.

Reply:
> What is the reason to change original fail-fast approach ?

Sorry, for that. Before, I thought it was a better idea but you're right, it should be done by previous fail-fast approach.